### ISSUE TITLE
Automatically grant encrypt to cloudtrail service prinicipal when using a kmskey

### ISSUE THREAD

**Issue by kevinvincent** [aws/aws-cdk#3822]
### General Information

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

### Description

I'd like the cloudtrail Trail construct to automatically grant Encrypt to the cloudtrail.amazonaws.com service principal when a kms key is passed into the trail construct. The problem is that currently theres no documentation that this grant has to exist and the error that CFN throws when running is super unhelpful:

AuditTrailFF0F2BB5 | CREATE_FAILED | Insufficient permissions to access S3 bucket...

Basically I'd like to add these lines in the Trail construct:
```
if (props.kmsKey) {
  props.kmsKey.grantEncrypt(new iam.ServicePrincipal("cloudtrail.amazonaws.com"))
}
```

### Environment

  - **CDK CLI Version:** 1.6
  - **Module Version:** 1.6
  - **OS:** all
  - **Language:** all

**Comment by portswigger-tim** [comment]
A potential issue might be that users might want to add conditions to the key policy, for example limiting `aws:SourceArn` to prevent unrestricted access to encrypt, and upload data to the configured bucket.

**Comment by TheRealAmazonKendra** [comment]
A better error message here might be the better path forward.
